rs/bitcoin/kyt/src/main.rs

```diff
@@ -92,7 +95,15 @@ fn init(arg: KytArg) {
 #[ic_cdk::post_upgrade]
 fn post_upgrade(arg: KytArg) {
     match arg {
-        KytArg::UpgradeArg(_) => (),
+        KytArg::UpgradeArg(arg) => {
```

Comment:
Is that something that should be changed by upgrade args instead of for example requiring the canister to be re-installed from scratch? Otherwise I think we should check that,

1. if a transaction was accepted when `KytMode::AcceptAll`, then upgrading the canister to `KytMode::Normal` and then calling `check_transaction` on the same transaction should trigger some HTTPs outcall (previous result was not cached).
2. Same thing if the initial state was `KytMode::RejectAll`

Reply:
Again, it was mostly because the existing ckbtc tests rely on upgrades.

The newly added tests checked the following two cases (which would imply that the cache is reset after upgrade):
1. A previously accepted address is now blocked after setting to `RejectAll`.
2. A previously blocked address is not accepted after setting to `AcceptAll`.

I can extend the test to addtionally check that after setting back to `Normal`, the "normal" behavior is restored. WDYT?

Reply:
Sorry, I mean to say `now accepted`. Will add the tests. Thanks!